### ISSUE TITLE
[Bug]: Incorrect Accessibility Role for Text Inside Buttons Causes a11y Test Failures

### ISSUE THREAD

**Issue by georgewrmarshall** [MetaMask/metamask-mobile#8630]
### Describe the bug

We've encountered an issue where text elements nested inside button components are incorrectly assigned the accessibility role of “text”. This misconfiguration leads to failures in accessibility (a11y) tests, which expect to find an accessible element with the role of "button" containing the specified text. The current setup hinders the ability of screen readers and other assistive technologies to correctly interpret and interact with our buttons, impacting the overall accessibility of our application.

### Issue Details
Accessibility tests designed to locate buttons based on their text content are failing because the text within buttons does not inherit or share the button's accessible role. Instead, the text is being identified with a separate and inappropriate role (“text”), making it difficult for the tests (and potentially for users of assistive technologies) to recognize the button as a clickable action element.

### Suggested Improvement
To resolve this issue, we need to ensure that text within buttons is correctly associated with the button's accessibility role, without being assigned a distinct role that could mislead or confuse accessibility tools and users. This might involve:
- Reviewing and updating the way we structure our buttons and text content within our codebase.
- Ensuring that any text nested inside a button is semantically and programmatically recognized as part of the button element.
- Verifying through a11y testing that buttons with nested text are correctly identified based on their text content.

### Acceptance Criteria
- Text elements within buttons do not have a separate accessibility role from the button.
- Accessibility tests successfully identify buttons based on their text content, without encountering role-related failures.
- Screen readers and other assistive technologies correctly interpret and interact with buttons, recognizing the text as part of the actionable button element.

### Expected behavior

- When conducting accessibility tests or using assistive technologies, the system should correctly identify and interact with buttons based on their text content.
- Text nested inside buttons should not be assigned a separate accessibility role but should be recognized as part of the button element. This ensures that the entire button, including its text, is perceived as a single actionable element.
- Accessibility tests looking for buttons based on text content should pass without errors, indicating that the button elements are correctly marked up and accessible.

### Screenshots/Recordings

![image (2)](https://github.com/MetaMask/metamask-mobile/assets/8112138/86556fb5-0fbc-482e-9199-714b894c974f)
![image](https://github.com/MetaMask/metamask-mobile/assets/8112138/9d43f4ea-0c3f-4f65-a318-fd8b331a1026)

### Steps to reproduce

1. Navigate to a page or component where a button contains text, for example, a submit button with the label "Submit".
2. Run accessibility (a11y) tests that check for elements with the role of "button" based on their text content. Tools like Axe, Lighthouse, or manual testing with a screen reader could be used.
3. Observe the test results, particularly noting any failures to identify buttons based on their text content.

### Version

7.15.0

### Build type

None

### Device

iPhone

### Operating system

iOS

### Severity

- Sev 2

**Comment by georgewrmarshall** [comment]
@wachunei please add correct steps to reproduce

**Comment by wachunei** [comment]
I have more details about the issue and an example to reproduce.

The issue is: Button components base component `ButtonBase` does not have default props for accessibility, these are `accessibilityRole: 'button'` and `accessible: true`. 

Currently a simplified snapshot of every Button of the component library would look like this:

```jsx
<TouchableOpacity>
  <Text
    accessibilityRole="text"
  >
    Click me!
  </Text>
</TouchableOpacity>
```

This won't work with tests like `screen.getByRole('button', { name: 'Click me!' })` because the button does not have the props. 

This is a working example snapshot:
```jsx
<TouchableOpacity
  accessibilityRole="button"
  accessible={true}
>
  <Text
    accessibilityRole="text"
  >
    Click me!
  </Text>
</TouchableOpacity>
```

In the branch [`temp/button-a11y-tests`](https://github.com/MetaMask/metamask-mobile/tree/temp/button-a11y-tests) I have added this file [`app/component-library/components/Buttons/Button/foundation/ButtonBase/ButtonBase.rntl.test.tsx`](https://github.com/MetaMask/metamask-mobile/blob/temp/button-a11y-tests/app/component-library/components/Buttons/Button/foundation/ButtonBase/ButtonBase.rntl.test.tsx) with examples of failing and passing tests.

The file includes a comparison with the deprecated `StyledButton` component that behaves correctly regarding a11y tests. 
 The tests include snapshots to compare the default rendering of the props missing https://github.com/MetaMask/metamask-mobile/blob/temp/button-a11y-tests/app/component-library/components/Buttons/Button/foundation/ButtonBase/__snapshots__/ButtonBase.rntl.test.tsx.snap
